src/halbana/halbana.validator.ts

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable no-unused-vars */
+import { Request, Response, NextFunction } from 'express';
+import { Types } from 'mongoose';
+import { InvalidField, NotExist, EmptyField } from '../errors/ClientErrors';
+
+export default class HalbanaValidator {
+
+    static checkName(req: Request, res: Response, next: NextFunction) {
+        if(req.body.name.length > 1) {
+            return next();
+        }
+        return next(new InvalidField('file name must include more than one letter.'));
+    }
+
+    static allFieldsValidation(req: Request, res: Response, next: NextFunction) {
+        if (req.body.name) {
+            if (req.body.deployed) {
+                    if (req.body.version) {
+                        return next();
+                    }
+                    return next(new EmptyField('Please enter a version field.'));
+            }
+            return next(new EmptyField('Please enter a boolean deployed field.'));
+        }
+        return next(new EmptyField('Please enter a name field.'));
```

Comment:
Avoid nesting code when it's not necessary.

```suggestion
        if (! req.body.name) return next(new EmptyField('Please enter a name field.'));
        if (!req.body.deployed) return next(new EmptyField('Please enter a boolean deployed field.'));
        if (!req.body.version) return next(new EmptyField('Please enter a version field.'));
        return next();                  
```

Using `not` can make the code `flat` and not nested, which can make it much easier to read and understand.